### PR TITLE
feat(data-table): add new prop verticalCellAlignment

### PR DESCRIPTION
**Pull request by jonnybel**
#### Summary

`DataTable`: This PR adds a prop to control the vertical alignment of the content of all data-cells (thus, excluding Headers).

Because there was already a prop called `cellAlignment`, it was renamed to `horizontalCellAlignment`, and the new prop is called `verticalCellAlignment`.

I'm not considering this a breaking change given the component's  "Beta" status (which is soon to end, as we're done with most of the new features).

### CODE REVIEW

File: packages/components/data-table/src/cell.js [hunk 1]
```diff
@@ -196,15 +197,18 @@ DataCell.defaultProps = {
...
+    >
       {props.children}
     </CellInner>
   </BaseFooterCell>
 );
 FooterCell.displayName = 'FooterCell';
 FooterCell.propTypes = {
   children: PropTypes.node.isRequired,
-  alignment: PropTypes.oneOf(['left', 'center', 'right']),
+  horizontalCellAlignment: PropTypes.oneOf(['left', 'center', 'right']),
```
Comment (tdeekens) [inline packages/components/data-table/src/cell.js:211]: Are or shoudl we defaulting this?

Comment (jonnybel) [reply packages/components/data-table/src/cell.js:211]: Yeah, forgot to add the default on the Footer.

File: packages/components/data-table/src/cell.js [hunk 2]
```diff
@@ -212,5 +213,8 @@ FooterCell.propTypes = {
   isCondensed: PropTypes.bool,
   numberOfColumns: PropTypes.number.isRequired,
 };
+FooterCell.propTypes = {
```
Comment (tdeekens) [inline packages/components/data-table/src/cell.js:216]: Get a ☕ 😄 

```suggestion
FooterCell.defaultProps = {
```

Comment (jonnybel) [reply packages/components/data-table/src/cell.js:216]: 🙈

Comment (jonnybel) [reply packages/components/data-table/src/cell.js]: `fix: remove stupidity`

**Pull request merged**